src/verifier.cc

```diff
@@ -15,15 +15,153 @@
 #include "src/verifier.h"
 
 #include <cmath>
+#include <vector>
+
+#include "src/command.h"
 
 namespace amber {
 namespace {
 
-const float kEpsilon = 0.002f;
+const double kEpsilon = 0.000001;
+const double kDefaultTexelTolerance = 0.002;
+
+// It returns true if the difference is within the given error.
+// If |is_tolerance_percent| is true, the actual tolerance will be
+// relative value i.e., |tolerance| * fabs(real - expected).
+// Otherwise, this method uses the absolute value i.e., |tolerance|.
+bool IsEqualWithTolerance(const double real,
+                          const double expected,
+                          double tolerance,
+                          const bool is_tolerance_percent = true) {
+  double difference = std::fabs(real - expected);
+  if (difference == 0.0)
+    return true;
+  if (is_tolerance_percent)
+    tolerance *= difference;
+  return difference < tolerance;
+}
 
-bool IsFloatPixelEqualInt(float pixel, uint8_t expected) {
-  // TODO(jaebaek): Change kEpsilon to tolerance.
-  return std::fabs(pixel - static_cast<float>(expected) / 255.0f) < kEpsilon;
+template <typename T>
+Result CheckValue(const ProbeSSBOCommand* command,
```

Comment:
Can these verifier changes be pulled out to a separate Cl and committed with the tests?